### PR TITLE
Create deployment script for staging deployments

### PR DESCRIPTION
**Pull request by marcellp**

### CODE REVIEW

File: biomage/stage/stage.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    required=True,
+    help="A GitHub Personal Access Token with the required permissions.",
+)
+@click.option(
+    "--org",
+    envvar="GITHUB_BIOMAGE_ORG",
+    default="biomage-ltd",
+    help="The GitHub organization to perform the operation in.",
+)
+def stage(token, org, refs):
```
Comment (ivababukova) [inline biomage/stage/stage.py:225]: what does `refs` stand for? Is it actually pinned deployments?

File: biomage/stage/stage.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    s = requests.Session()
+    r = s.get(url)
+
+    Deployment = namedtuple("Deployment", ["ref", "url", "status", "text"])
+
+    return Deployment(ref=ref or "master", url=url, status=r.status_code, text=r.text)
+
+
+def compile_requirements(org, refs):
+    DEPLOYMENTS = ("ui", "api", "worker")
```
Comment (ivababukova) [inline biomage/stage/stage.py:39]: these make more sense to be called `repos`, rather than deployments

File: biomage/stage/stage.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+    ref_data = {deployment: None for deployment in DEPLOYMENTS}
```
Comment (ivababukova) [inline biomage/stage/stage.py:41]: same here: it should be `repo` and `repos`

**Comment by ivababukova** [comment]
You also need to write a README, giving instructions on how people can create a staging environments. Can you also put instructions in the README about the other things that we can do in this repo - about configure_repo and rotate_ci?

File: biomage/stage/stage.py (hunk 1 above)
Comment (marcellp) [reply biomage/stage/stage.py:225]: `refs` are the repositories and their versions that need to be deployed in a staging sandbox. For example, `api` is a ref, and by default corresponds to the repository `api` on `master`. `ui/5` is a ref, and it corresponds to the repository `ui` on the branch for pull request `5`.

File: biomage/stage/stage.py (hunk 2 above)
Comment (marcellp) [reply biomage/stage/stage.py:39]: Fair point.

File: biomage/stage/stage.py (hunk 3 above)
Comment (marcellp) [reply biomage/stage/stage.py:41]: Fair point.

File: biomage/stage/stage.py (hunk 1 above)
Comment (ivababukova) [reply biomage/stage/stage.py]: I see, I got confused then!

**Comment by marcellp** [comment]
> You also need to write a README, giving instructions on how people can create a staging environments. Can you also put instructions in the README about the other things that we can do in this repo - about configure_repo and rotate_ci?

Added a README. Let me know if this is good enough.

**Review by ivababukova — APPROVED** [review]

**Pull request merged**